python-frontend/src/main/java/org/sonar/plugins/python/api/tree/BaseTreeVisitor.java

```diff
@@ -561,4 +561,19 @@ public void visitKeywordPattern(KeywordPattern keywordPattern) {
   public void visitValuePattern(ValuePattern valuePattern) {
     scan(valuePattern.qualifiedExpression());
   }
+
+  @Override
+  public void visitLineMagic(LineMagic lineMagic) {
+    // noop
+  }
+
+  @Override
+  public void visitLineMagicStatement(LineMagicStatement lineMagicStatement) {
+    // noop
+  }
+
+  @Override
+  public void visitCellMagicStatement(CellMagicStatement lineMagicStatement) {
+    // noop
+  }
```

Comment:
I'm not sure if it's a good idea exposing `IPython` nodes in the public API for the time being, especially because we don't intend to use them for anything useful so far.
I would instead suggest to:
- either remove those strongly typed node from the AST as long as we don't need them
- or create a private `IPythonTree` extending `Tree` which contains the `IPython` specific visitor, nodes, ...

Reply:
As we discussed, we can do without the visitors. We also can flag the interfaces for the strongly typed nodes as `Beta`. It's still useful to have them (for tests, for example), and we suspect we'll need them in the future, so if that's okay, I'd suggest keeping them as `Beta` for now.